plugins/chat/app/services/chat/channel_view_builder.rb

```diff
@@ -50,6 +50,7 @@ class Contract
       attribute :direction, :string # (optional)
       attribute :page_size, :integer # (optional)
       attribute :fetch_from_last_read, :boolean # (optional)
+      attribute :target_date, :string # (optional)
```

Comment:
Actually this service is a better place to validate that this is a correct date/time -- we need to add a `validate target_date_ok?` method to validate it. It's the same as rails model validations e.g. 

https://github.com/discourse/discourse/blob/a4630d220b55c45b9b72cbc1e598c0d626f2e598/app/models/invite.rb#L39-L39

https://github.com/discourse/discourse/blob/a4630d220b55c45b9b72cbc1e598c0d626f2e598/app/models/invite.rb#L70-L72

Reply:
I think date is very hard to validate because it can be a string or anything with can be casted to string. What validation would you write?

Reply:
Hmm I guess you are right, I had assumed we would have a pattern for this. Fair enough to leave it out.